src/nspawn/nspawn.c

```diff
@@ -4089,7 +4149,24 @@ static int outer_child(
         if (r < 0)
                 return r;
 
-        r = copy_devnodes(directory);
+        /* FUSE is only userns-safe in FUSE version 7.27 and later.
+         * https://github.com/torvalds/linux/commit/da315f6e03988a7127680bbc26e1028991b899b8 */
+        uint32_t fuse_major, fuse_minor;
+        r = get_fuse_version(&fuse_major, &fuse_minor);
+        if (r < 0)
+                log_warning("Disabling FUSE: Failed to determine FUSE version");
+        else if (r > 0)
```

Comment:
hmmm, i'd turn around the return values here. for functions that return three relevant cases – i.e. failure, success without doing anything, and success with doing something – we usually map these to < 0, == 0, > 0, in this order. i.e. you can consider the success case a boolean if so will that indicates whether we did something or just kipped over it.

Hence, to follow this pattern, I think get_fuse_version() should return 0 as retval and 0/0 as major/minor in case FUSE is absent. And it should return 1 (with major/minor filled in properly) in case it is present.

Reply:
Done.